### PR TITLE
Added Windows support

### PR DESCRIPTION
**Pull request by TimHacker**
I've borrowed heavily from the previous PR #11 and have just made a few tweaks based on the feedback on there and also to make it play nicely with posh-git/post-hg.

Let me know if you have any edits that you'd like me to make.

Thanks,
Tim

### CODE REVIEW

**Comment by haustraliaer** [comment]
I've been giving this a go as a local plugin - only issue I found was I can't open a new tab when a process (such as webpack) is running - get the following error:

image

File: index.js [hunk 1]
```diff
@@ -4,26 +4,47 @@ let curPid;
...
+          curCwd = cwd;
+        }
+      }
+    });
+  }
+};
 
 exports.middleware = (store) => (next) => (action) => {
   switch (action.type) {
+    case 'SESSION_ADD_DATA':
```
Comment (hharnisc) [inline index.js]: Curious why this can't be done in the session SESSION_PTY_DATA?

File: index.js [hunk 2]
```diff
@@ -4,26 +4,47 @@ let curPid;
...
+      if (process.platform === 'win32' && directoryRegex.test(action.data)) {
```
Comment (hharnisc) [inline index.js]: Could you talk me through why this works? Does windows output the directory on every command or something like that?

**Comment by hharnisc** [comment]
Fixed this one in #25 - thank you for the PR @TimHacker

**Pull request closed without merging**